src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -129,20 +176,51 @@ export class MarkdownToolbarComponent {
 			{ action: this._toggleSplitViewAction },
 			{ action: this._toggleMarkdownViewAction }
 		];
+
+		this._previewModeTaskbarContent = [
+			{ action: boldButton },
+			{ action: italicButton },
+			{ action: underlineButton },
+			{ action: highlightButton },
+			{ action: codeButton },
+			{ element: linkButtonContainer },
+			{ action: listButton },
+			{ action: orderedListButton },
+			{ element: imageButtonContainer },
+			{ element: buttonDropdownContainer },
+			{ action: this._toggleTextViewAction },
+			{ action: this._toggleSplitViewAction },
+			{ action: this._toggleMarkdownViewAction }
+		];
+
 		// Hide link and image buttons in WYSIWYG mode
 		if (this.cellModel.showPreview && !this.cellModel.showMarkdown) {
 			this._actionBar.setContent(this._wysiwygTaskbarContent);
 		} else {
-			this._actionBar.setContent(this._taskbarContent);
+			if (this.previewFeaturesEnabled) {
+				this._actionBar.setContent(this._previewModeTaskbarContent);
+			} else {
+				this._actionBar.setContent(this._taskbarContent);
+			}
 		}
 	}
 
+	public onCustomButtonClick(event, type: MarkdownButtonType) {
```

Comment:
Name isn't super clear - maybe something like `onTransformTextButtonClick`

Reply:
TransformText is reserved for transformations that wrap existing text with tags: (e.. Bold, Italics, etc). These two are buttons used for inserting elements. I'll go with: onInsertButtonClick